homeassistant/components/goodwe/services.yaml

```diff
@@ -0,0 +1,33 @@
+set_work_mode:
+  description: |
+    Set the inverter work_mode.
+    (Use with caution, it modifies inverter's operational parameter, usually accessible to installers only.)
+  fields:
+    entity_id:
+      description: ID of the inverter entity
+      example: "sensor.goodwe_inverter"
+    work_mode:
+      description: 0 - General mode, 1 - Off grid mode, 2 - Backup mode, 3 - Wait mode
```

Comment:
This service could be a select entity instead, or a number entity. But select seems more user friendly since the numbers have a meaning in words.

Reply:
Sorry for stupid question, just want to make sure I understand it properly.
You suggest not to have one entity with 3 services, but rather 3 (specialized) entities instead, right ?

Reply:
Yes, I think there should be three entities but on one device (inverter).

Reply:
The select entity allows the user to select from a set of options.

Reply:
Understand, will change that ...

Reply:
Done, it was replaced with select entity, which is really way better. Please review.